mechanisms/default/hh.mod

```diff
@@ -13,83 +12,74 @@ UNITS {
 }
 
 PARAMETER {
-    gnabar = .12 (S/cm2)
-    gkbar = .036 (S/cm2)
-    gl = .0003 (S/cm2)
-    el = -54.3 (mV)
-    celsius (degC)
+    gnabar =   0.12   (S/cm2)
+    gkbar  =   0.036  (S/cm2)
+    gl     =   0.0003 (S/cm2)
+    el     = -54.3    (mV)
+    celsius           (degC)
 }
 
-STATE {
-    m h n
-}
-
-ASSIGNED {
-    v (mV)
+STATE { m h n }
 
-    gna (S/cm2)
-    gk (S/cm2)
-    minf
-    hinf
-    ninf
-    mtau (ms)
-    htau (ms)
-    ntau (ms)
-    q10
-}
+ASSIGNED { q10 }
 
 BREAKPOINT {
     SOLVE states METHOD cnexp
-    gna = gnabar*m*m*m*h
-    ina = gna*(v - ena)
-    gk = gkbar*n*n*n*n
-    ik = gk*(v - ek)
-    il = gl*(v - el)
+          
+    ina = gnabar*m*m*m*h*(v - ena)
```

Comment:
For these calculations, I would prefer to still calculate `gna` and `gna` as `LOCAL` variables

We evaluate the expression `ik = g*(v - ek)` to determine the conductance `g`, which gives the following in the generated C++ code for the original version:
```C++
conductivity_ = gna[i_]+gk[i_]+gl[i_]
```
What does the new version give?

Reply:
I use the SIMD version here
```
assign(conductivity_, S::add(S::add(S::mul(S::mul(S::mul(S::mul(simd_cast<simd_value>(indirect(gnabar+index_, simd_width_)), simd_cast<simd_value>(indirect(m+index_, simd_width_))), simd_cast<simd_value>(indirect(m+index_, simd_width_))), simd_cast<simd_value>(indirect(m+index_, simd_width_))), simd_cast<simd_value>(indirect(h+index_, simd_width_))), simd_cast<simd_value>(indirect(gl+index_, simd_width_))), S::mul(S::mul(S::mul(S::mul(simd_cast<simd_value>(indirect(gkbar+index_, simd_width_)), simd_cast<simd_value>(indirect(n+index_, simd_width_))), simd_cast<simd_value>(indirect(n+index_, simd_width_))), simd_cast<simd_value>(indirect(n+index_, simd_width_))), simd_cast<simd_value>(indirect(n+index_, simd_width_)))));
```
so simplified
```
conductivity_ = add(add(mul(mul(mul(mul(indirect(gnabar+index_)), indirect(m+index_))), indirect(m+index_))), indirect(m+index_))), indirect(h+index_))), indirect(gl+index_))), mul(mul(mul(mul(indirect(gkbar+index_)), indirect(n+index_))), indirect(n+index_))), indirect(n+index_))), indirect(n+index_)))));
```
and this is
```
gnabar*m^3*h + gl + gkbar * n^4
```
which is the same as in the original.

I do not see how a LOCAL would help. What I would expect to help is storing and `m` and `n` into LOCALs to avoid multiple `load`s. That is if the compiler does not figure it out on its own.

Reply:
Good point, quite an improvement.